README.md

```diff
@@ -1,801 +1,3854 @@
+   ```
+1. ```
+    $ export GOPATH=${HOME}/BRUCE
+   ```
+1. ```
+    $ export GOBIN=$GOPATH/bin
+   ```
+1. ```
+    $ export GO111MODULE=on
+   ```
+1. ```
+    $ export GOROOT=/usr/local/go
+   ```
+1. ```
+    $  export PATH=$PATH:${GOROOT}/bin 
+   ```
+
+
+## Configuring Docker proxy
+<blockquote>
+NOTE: Before performing the following steps, ensure the `http_proxy`, `https_proxy`, and `no_proxy` environment variables are set.
+</blockquote>
+1. [Optional] If the following content is not present in the `/etc/environment` file, add it:
+   
+	```
+    $ cat << EOF | sudo tee -a /etc/environment
+    http_proxy=${http_proxy}
+    https_proxy=${https_proxy}
+    no_proxy=${no_proxy}
+	EOF
+	```
+2. Run the following commands to update proxy information in the Docker service file:
+   ```
+   $ sudo mkdir -p /etc/systemd/system/docker.service.d
+   ```
+   
+   ```
+   $ cat << EOF | sudo tee /etc/systemd/system/docker.service.d/http-proxy.conf
+    [Service]
+    Environment="HTTP_PROXY=${http_proxy}"
+    Environment="HTTPS_PROXY=${https_proxy}"
+    Environment="NO_PROXY=${no_proxy}"
+    EOF
+   ```
+3. Run the following commands to update proxy information in the user Docker config file:
+   ```
+   $ mkdir ~/.docker
+   ```
+   
+   ```
+   $ sudo chown ${USER}:${USER} ~/.docker -R
+   ```
+   
+   ```
+   $ sudo chmod 0700 ~/.docker
+   ```
+   
+   ```
+   $ cat > ~/.docker/config.json <<EOF
+    {
+     "proxies":
+     {
+      "default":
+      {
+      "httpProxy": "${http_proxy}",
+      "httpsProxy": "${https_proxy}",
+      "noProxy": "${no_proxy}"
+      }
+    }
+   }
+   EOF
+   ```
+   
+## Installing Docker
+
+1. Run the following commands:
+   
+   1. ```
+      $ sudo apt-get install -y apt-transport-https=1.6.12ubuntu0.2 ca-certificates=20210119~18.04.1 curl=7.58.0-2ubuntu3.12
+      ```
+	  
+   2. ```
+      $ sudo apt-get install -y gnupg-agent=2.2.4-1ubuntu1.3 software-properties-common=0.96.24.32.14
+      ```
+	  
+   3. ```
+      $ curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+      ```
+	  
+   4. ```
+      $ sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+      ```
+	  
+   5. ```
+      $  sudo apt-get install -y docker-ce=5:19.03.12~3-0~ubuntu-bionic docker-ce-cli=5:19.03.12~3-0~ubuntu-bionic containerd.io --allow-downgrades
+      ```
+
+2. Configure overlay storage for Docker:
+   ```
+   $ cat << EOF | sudo tee /etc/docker/daemon.json
+     {
+       "exec-opts": ["native.cgroupdriver=systemd"],
+       "log-driver": "json-file",
+       "storage-driver": "overlay2"
+     }
+     EOF
+   ```
+3. Perform the following steps to check and create Docker group if doesn't exist:
+
+   1. Check if the Docker group exists using the following command:
+      ```
+	    $ getent group docker
+	   ```
+	  ```
+	
+     ```
+   
+	  Example output: `docker:x:998:<username>`
```

Comment:
output is not showing username:
bruce@dnode1:~$ getent group docker
docker:x:999:
bruce@dnode1:~$